### PR TITLE
[Fix](multi-catalog) Fix query hms tbl with compressed data files.

### PR DESCRIPTION
**Pull request by dutyu**
# Proposed changes

Issue Number: close #xxx

## Problem summary

If a hms table's file format is csv, uncompressed data files may be coexists with compressed data files, so we need to set compressType separately.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by dutyu** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.06 seconds
 stream load tsv:          421 seconds loaded 74807831229 Bytes, about 169 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          61 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          34 seconds loaded 861443392 Bytes, about 24 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230511113047_clickbench_pr_142344.html

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java [hunk 1]
```diff
@@ -242,9 +243,19 @@ public void createScanRangeLocations() throws UserException {
...
             FileSplit fileSplit = (FileSplit) split;
 
+            TFileScanRangeParams scanRangeParams;
+            if (!isCsvFormat) {
+                scanRangeParams = params;
+            } else {
+                // If fileFormatType is csv format, uncompressed files may be coexists with compressed files
+                // So we need set compressType separately
+                scanRangeParams = new TFileScanRangeParams(params);
+                scanRangeParams.setCompressType(getFileCompressType(fileSplit));
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/FileQueryScanNode.java:255]: I think we should do this for json, too

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Pull request merged**